iroh-gateway/src/headers.rs

```diff
@@ -74,12 +74,41 @@ pub fn add_content_disposition_headers(
 
 #[tracing::instrument()]
 pub fn set_content_disposition_headers(headers: &mut HeaderMap, filename: &str, disposition: &str) {
+    // TODO: handle non-ascii filenames https://github.com/ipfs/specs/blob/main/http-gateways/PATH_GATEWAY.md#content-disposition-response-header
     headers.insert(
         CONTENT_DISPOSITION,
         HeaderValue::from_str(&format!("{}; filename={}", disposition, filename)).unwrap(),
     );
 }
 
+#[tracing::instrument()]
+pub fn add_content_range_headers(headers: &mut HeaderMap, range: Range<u64>, size: Option<u64>) {
+    let content_range = if let Some(size) = size {
+        format!("bytes {}-{}/{}", range.start, range.end - 1, size)
```

Comment:
if `range.end`  is `0` this will panic

Reply:
I can add another guard, but this path is not invoked unless range exists and is passed in. And that is checked against in `parse_range_header`

Reply:
then maybe add a comment about that